Sources/MarathonCore/Export.swift

```diff
@@ -0,0 +1,165 @@
+
+        try perform(export(tempFile, to: destination), orThrow: Error.failedToExportScript(tempFile.name))
+        try tempFile.delete()
+    }
+
+    private func destinationFolder() throws -> Folder {
+        let folder: Folder
+        if arguments.count > 1 && !arguments[1].contains("--force") {
+            folder = try FileSystem().createFolderIfNeeded(at: arguments[1])
+        } else {
+            folder = FileSystem().currentFolder
+        }
+        return folder
+    }
+
+    private func overwriteIfNeeded(destination: Folder, scriptName: String) throws {
+        let exportPath = destination.path.appending(scriptName)
+        let existingProjectFolder = try? destination.subfolder(atPath: scriptName)
+
+        if existingProjectFolder != nil && !arguments.contains("--force") {
+            printer.output("⚠️  A directory already exists at \(exportPath)")
+            printer.output("❓  Are you sure you want to overwrite it? (Type 'Y' to confirm)")
+            if readLine()?.lowercased() == "n" {
+                exit(1)
+            }
+        }
+        try existingProjectFolder?.delete()
+    }
+
+    private func export(_ file: File, to folder: Folder) throws {
+        let projectFolder = try folder.createSubfolder(named: file.scriptName)
+
+        let packageFile = try projectFolder.createFile(named: "Package.swift")
+        let packages = try resolvePackages(from: file)
+        let packageFileString = try makePackageFileString(for: file, with: packages)
+        try packageFile.write(string: packageFileString)
+
+        let sourcesFolder = try projectFolder.createSubfolder(named: "Sources")
+        let scriptFileString = try makeFileStringWithInlineDependencies(for: file, using: packages)
+        let scriptFile = try sourcesFolder.createFile(named: file.name)
+        try scriptFile.write(string: scriptFileString)
+    }
+
+    private func resolvePackages(from file: File) throws -> [Package] {
+        let importNames = try file.importNames()
+        let allManagedPackages = packageManager.addedPackages
+        let scriptPackages: [Package] = importNames.flatMap { name in
+            if let scriptPackage = allManagedPackages.first(where: { $0.name.lowercased() == name.lowercased() }) {
+                return scriptPackage
+            } else {
+                return nil
+            }
+        }
+        return scriptPackages
+    }
+
+    private func makePackageFileString(for file: File, with packages: [Package]) throws -> String {
+        let base = "import PackageDescription\n"
+            + "\n"
+            + "let package = Package(\n"
+            + "    name: \"\(file.scriptName)\",\n"
+            + "    dependencies: [\n"
+
+        let baseWithPackages = packages.reduce(base) { partialResult, package in
+            let string = "        .Package(url: \"\(package.url)\", majorVersion: \(package.majorVersion)),\n"
```

Comment:
This will fail on our Swift 4 build (which you should get if you rebase against `master`), since the package manifest API has changed. I think we should extract part of the code from `generateMasterPackageDescription(forSwiftToolsVersion:)` to have a single place where we construct package manifests. Let me know if you want my help on this 🙂